log_forwarder/parser.py

```diff
@@ -22,68 +23,72 @@ def parse(self, line: str):
             return json.dumps(parsed.groupdict())
         return line
 
-    def parse_file(self, logfile):
-        parse_filepath = os.path.join(os.path.dirname(logfile.get_filepath()), "parsed_" +
-                                      os.path.basename(logfile.get_filepath())) + ".gz"
-        with gzip.open(parse_filepath, 'wb') as f:
-            for line in logfile.get_lines():
+    def parse_file(self):
+        parse_filepath = os.path.join(os.path.dirname(self.input_file.get_filepath()), "parsed_" +
+                                      os.path.basename(self.input_file.get_filepath())) + ".gz"
+        with gzip.open(parse_filepath, 'wt') as f:
+            for line in self.input_file.get_lines():
                 parsed = self.parse(line) + "\n"
-                f.write(parsed.encode())
+                f.write(parsed)
         return parse_filepath
 
+    def get_parsed_lines(self):
+        for line in self.input_file.get_lines():
+            yield self.parse(line)
+
 
 class S3AccessLogParser(Parser):
 
     REGEX = r'(?P<BucketOwner>[^ ]*) (?P<Bucket>[^ ]*) \[(?P<RequestDateTime>.*?)\] (?P<RemoteIP>[^ ]*) (?P<Requester>[^ ]*) (?P<RequestID>[^ ]*) (?P<Operation>[^ ]*) (?P<Key>[^ ]*) (\"(?P<RequestURI_operation>[^ ]*) (?P<RequestURI_key>[^ ]*) (?P<RequestURI_httpProtoversion>- |[^ ]*)\"|\"-\"|-) (?P<HTTPstatus>-|[0-9]*) (?P<ErrorCode>[^ ]*) (?P<BytesSent>[^ ]*) (?P<ObjectSize>[^ ]*) (?P<TotalTime>[^ ]*) (?P<TurnAroundTime>[^ ]*) (\"(?P<Referrer>.*?)\"|-) (\"(?P<UserAgent>.*?)\"|-) (?P<VersionId>[^ ]*)(?: (?P<HostId>[^ ]*) (?P<SigV>[^ ]*) (?P<CipherSuite>[^ ]*) (?P<AuthType>[^ ]*) (?P<EndPoint>[^ ]*) (?P<TLSVersion>[^ ]*))?( (?P<S3AccessPointARN>arn:[^ ]*|-))?'
```

Comment:
Are these regex hand crafted or from aws? might be worth a ref link if they are from aws

Reply:
That doesn't show in the PR but the link to where they come from is referenced at the top of this file: https://github.com/logchatio/brontobytes-aws-ingestion-python/pull/24/files#diff-369d9547f4e09ffc7d5443542126bd532ed336dd0f9ade58ed701dd90bde9617R6